tests/test_surface/test_regular_surface.py

```diff
@@ -1092,3 +1092,59 @@ def test_smoothing():
     assert mean2 == pytest.approx(1698.65, abs=0.3)  # smoothed ~same mean
 
     assert mean1 != mean2  # but not exacly same
+
+
+@pytest.mark.skipif(
+    xtgeo.__version_tuple__[0] >= 4,
+    reason="Not relevant after deprecated __init__ is removed",
+)
+def test_loadvalues_before_remove(default_surface):
+    """Test that load_values() has the claimed effect before deprecating __init__"""
+    correct = xtgeo.RegularSurface(filesrc=TESTSET1, **default_surface)
+
+    srf = xtgeo.RegularSurface(filesrc=TESTSET1, **default_surface)
+    srf.load_values()
+    assert (correct == srf).all(), "Surface should not have been modified"
+
+    # Remove any "values"-parameter and let deprecated __init__ add one
+    default_surface.pop("values", None)
+    srf = xtgeo.RegularSurface(filesrc=TESTSET1, **default_surface)
+    srf.load_values()
+    assert (correct == srf).all(), "Surface should not have been modified"
+
+    # Also specify the format - see test for after-remove where it behaves differently
+    srf = xtgeo.RegularSurface(
+        filesrc=TESTSET1, fformat="irap_binary", **default_surface
+    )
+    srf.load_values()
+    assert (correct == srf).all(), "Surface should not have been modified"
+
+    # Finally remove any "filesrc"-parameter
+    default_surface.pop("filesrc", None)
+    srf = xtgeo.RegularSurface(**default_surface)
+    srf.load_values()
+    assert (correct == srf).all(), "Surface should not have been modified"
+
+
+@pytest.mark.skipif(
+    xtgeo.__version_tuple__[0] < 4,
+    reason="Not relevant before deprecated __init__ is removed",
+)
+def test_loadvalues_after_remove(default_surface):
+    default_surface.pop("values", None)
+    with pytest.raises(ValueError, match=r"Unknown file format None"):
+        srf = xtgeo.RegularSurface(filesrc=TESTSET1, **default_surface)
```

Comment:
Once the version is `>=4` initialization giving `filesrc` to the constructor will be gone, so its best to move directly to `surface_from_file` in this test. Also, I think parsing the version information should be done similarly to how its done it `test_deprecations.py`.

Reply:
Agreed.